### ISSUE TITLE
File permissions are set to '0600' for repos created and should be '0755'

### ISSUE THREAD

**Issue by joshoohaah** [kemra102/yumserver-cookbook#6]
I've noticed that repos created with this come up with `0600` access.  so Nginx shows the parent directories but then you can't actually access any of the RPM's the cookbook is hosting.

**Comment by joshoohaah** [comment]
the repo RPM's should be set to `755` not `600`

**Comment by joshoohaah** [comment]
seems this command in your library `reposync -d -c /etc/reposync.conf -r #{repo} -p #{path}"` makes the files, but in the wrong mode. not sure how to change that either at the moment.  except with maybe after the reposync is done doing somethign like this
```
execute 'changing the file permissions on all repos' do
  command "chmod -R 0755 #{node['yumserver']['basepath']}"
  action :run
end
```

**Comment by kemra102** [comment]
The code specifically sets directory permissions to `0755`. I just confirmed this is the case on our production systems.

In regards to the RPMs pulled down by `reposync` on our production systems they are `0644` so are displayed by NGINX just fine.

This may be an issue with your setup specifically, can you provide steps to reproduce the issues you see?

**Comment by joshoohaah** [comment]
It's very basic really.  It is hosted on a centos 6.7 box.  Here is the code.  I added that last part to change the permissions on all the files where the yumserver_mirror resource drops them
```
include_recipe 'yumserver'

yumserver_mirror 'oel6_latest' do
  repo_name 'oel6_latest'
  repo_description 'Yum Packages for Oracle Linux 6'
  repo_baseurl 'http://public-yum.oracle.com/repo/OracleLinux/OL6/latest/$basearch/'
  local_path '/yum'
  # local_path node['hc_yumserver']['yum_dir']
  action :create
  use_repo false
end

yumserver_mirror 'oel6_UEK' do
  repo_name 'oel6_UEK'
  repo_description 'Unbreakable Kernel for Oracle 6'
  repo_baseurl 'http://public-yum.oracle.com/repo/OracleLinux/OL6/UEKR3/latest/$basearch/'
  # local_path '/yum'
  local_path node['hc_yumserver']['yum_dir']
  action :create
  use_repo false
end

yumserver_mirror 'epel7' do
  repo_name 'epel7'
  repo_description 'Extra Packages for Enterprise Linux 7'
  repo_baseurl 'https://dl.fedoraproject.org/pub/epel/7/x86_64/'
  action :create
  use_repo false
end
yumserver_mirror 'epel6' do
  repo_name 'epel6'
  repo_description 'Extra Packages for Enterprise Linux 6'
  repo_baseurl 'https://dl.fedoraproject.org/pub/epel/6/x86_64/'
  action :create
  use_repo false
end

execute 'changing the file permissions on all repos' do
  command "chmod -R 0755 #{node['yumserver']['basepath']}"
  action :run
end
```

**Comment by joshoohaah** [comment]
side note. I've also noticed even if i set the `local_path '/yum` in the resource its still going to `/var/lib/yum-repo`.   Not the end of the world, as i just put my mount to look there instead of at where initially stubbed it at `/yum` but it didn't do as i was thinking it would.

**Comment by kemra102** [comment]
I've not been able to reproduce the issues you are seeing on CentOS 6.7

I did however fix the issue with the baserepo in 17dff296

I'll release a bugfix update when the tests have finished running for that one.

Back to the larger issue - unless we can come up with a way to re-produce the issue reliably you may have to continue using your workaround.

**Comment by kemra102** [comment]
Bug fix release for basepath issue: https://supermarket.chef.io/cookbooks/yumserver/versions/2.0.2
